scripts/cleanup-bcd-overrides.ts

```diff
@@ -0,0 +1,139 @@
+const compat = new Compat();
+
+const argv = yargs(process.argv.slice(2))
+  .scriptName("cleanup-bcd-overrides")
+  .usage(
+    "$0 [paths..]",
+    "Remove unneeded `compat_features` overrides from files",
+    (yargs) =>
+      yargs.positional("paths", {
+        describe: "Directories or files to check/update.",
+        default: ["features"],
+      }),
+  ).argv;
+
+const logger = winston.createLogger({
+  level: argv.verbose > 0 ? "debug" : "warn",
+  format: winston.format.combine(
+    winston.format.colorize(),
+    winston.format.simple(),
+  ),
+  transports: new winston.transports.Console(),
+});
+
+let exitStatus = 0;
+
+setLogger(logger);
+
+const tagsToFeatures: Map<string, Feature[]> = (() => {
+  // TODO: Use Map.groupBy() instead, when it's available
+  const map = new Map();
+  for (const feature of compat.walk()) {
+    for (const tag of feature.tags) {
+      let features = map.get(tag);
+      if (!features) {
+        features = [];
+        map.set(tag, features);
+      }
+      features.push(feature);
+    }
+  }
+  return map;
+})();
+
+/**
+ * Check that the installed @mdn/browser-compat-data (BCD) package matches the
+ * one pinned in `package.json`. BCD updates frequently, leading to surprising
+ * error messages if you haven't run `npm install` recently.
+ */
+function checkForStaleCompat(): void {
+  const packageBCDVersionSpecifier: string = (() => {
+    const packageJSON: unknown = JSON.parse(
+      fs.readFileSync(process.env.npm_package_json, {
+        encoding: "utf-8",
+      }),
+    );
+    if (typeof packageJSON === "object" && "devDependencies" in packageJSON) {
+      const bcd = packageJSON.devDependencies["@mdn/browser-compat-data"];
+      if (typeof bcd === "string") {
+        return bcd;
+      }
+      throw new Error(
+        "@mdn/browser-compat-data version not found in package.json",
+      );
+    }
+  })();
+  const installedBCDVersion = compat.version;
+
+  if (!packageBCDVersionSpecifier.includes(installedBCDVersion)) {
+    logger.error(
+      `Installed @mdn/browser-compat-data (${installedBCDVersion}) does not match package.json version (${packageBCDVersionSpecifier})`,
+    );
+    logger.error("Run `npm install` and try again.");
+    process.exit(1);
+  }
+}
+
+function cleanup(sourcePath: string): void {
+  const source = YAML.parseDocument(
+    fs.readFileSync(sourcePath, { encoding: "utf-8" }),
+  );
+  const { name: id } = path.parse(sourcePath);
+
+  // Collect tagged compat features. A `compat_features` list in the source
+  // takes precedence, but can be removed if it matches the tagged features.
+  const taggedCompatFeatures = (tagsToFeatures.get(`web-features:${id}`) ?? [])
+    .map((f) => `${f.id}`)
+    .sort();
+
+  const data = source.contents.get("compat_features");
+
+  if (data) {
+    const features = data.items.map((item) => item.value).sort();
+    if (isDeepStrictEqual(features, taggedCompatFeatures)) {
+      source.contents.delete("compat_features");
```

Comment:
This will also delete comments that are connected to this `compat_features` source, plus any attached to the items of the array. The yaml library does make it possible to preserve these comments, but it's a little bit of a fuss. Options here would be to (in order of difficulty of implementation):

- Deal with it, making sure not to delete comments on review when this script runs. This is OK, but we wouldn't be able to assume the results of the script are safe to use.
- Error out if comments exist on the relevant parts of the document. A zero exit status would mean it's safe to use, non-zero would require human handling.
- Move comments to earlier nodes in the document. No comments would ever be lost, so the script's output would be very low risk.

See also:

- https://github.com/web-platform-dx/web-features/pull/2334#pullrequestreview-2462350349
- https://eemeli.org/yaml/v1/#comments

Reply:
I attempted to preserve the comments programmatically, but it doesn't seem like the comments are always parsed and registered correctly.  Some comments were still getting deleted, even though I made sure to check and utilize both comments-related variables.

I've looked through the diff the script produces, however, and I see that there's really only a handful of comments -- a small enough number that we can handle these cases manually.  I'll submit a new PR to take care of them!